google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/CloudEnv.java

```diff
@@ -81,6 +103,24 @@ private CloudEnv(
       dataSettings.stubSettings().setEndpoint(dataEndpoint);
     }
 
+    if (isDirectPathEnabled()) {
+      TransportChannelProvider channelProvider = dataSettings.stubSettings().getTransportChannelProvider();
+      InstantiatingGrpcChannelProvider defaultTransportProvider = (InstantiatingGrpcChannelProvider) channelProvider;
+      InstantiatingGrpcChannelProvider instrumentedTransportChannelProvider =
+          defaultTransportProvider
+              .toBuilder()
+              .setChannelConfigurator(
```

Comment:
I think it would be slightly cleaner to use `setInterceptorProvider` instead of the configurator

Reply:
Thanks for the pointer. One concern I had of using `setInterceptorProvider` is, if the existing channel provider already has its interceptor provider, then we would be replacing the existing one with the one we created. It would be good if we could retrieve the existing interceptor provider and add our own interceptor on top of that, but I couldn't find a good way of doing this. WDYT?

Reply:
Thats a fair point. It would be nice if gax had the ability to add an interceptor. I guess leave it as is for now